### PR TITLE
Fix EnrichSecurityIT resource cleanup

### LINKED ISSUE #84885: [CI] EnrichSecurityIT testBasicFlowDate failing
```
    org.elasticsearch.client.ResponseException: method [DELETE], host [http://[::1]:35575], URI [/_enrich/policy/my_policy], status line [HTTP/1.1 409 Conflict]	
    {"error":{"root_cause":[{"type":"status_exception","reason":"Could not delete policy [my_policy] because a pipeline is referencing it [my_pipeline]"}],"type":"status_exception","reason":"Could not delete policy [my_policy] because a pipeline is referencing it [my_pipeline]"},"status":409}	
        at app//org.elasticsearch.client.RestClient.convertResponse(RestClient.java:346)	
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:312)	
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:287)	
        at app//org.elasticsearch.test.enrich.CommonEnrichRestTestCase.deletePolicies(CommonEnrichRestTestCase.java:45)	
        at java.base@17.0.2/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	
        at java.base@17.0.2/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)	
 
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/ooruaod473f74/tests/:x-pack:plugin:enrich:qa:rest-with-security:javaRestTest/org.elasticsearch.xpack.enrich.EnrichSecurityIT/testBasicFlowDate

**Reproduction line:**
`./gradlew ':x-pack:plugin:enrich:qa:rest-with-security:javaRestTest' --tests "org.elasticsearch.xpack.enrich.EnrichSecurityIT.testBasicFlowDate" -Dtests.seed=14AEA6846D2CD5B -Dtests.locale=mk-MK -Dtests.timezone=America/Moncton -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.enrich.EnrichSecurityIT&tests.test=testBasicFlowDate

**Failure excerpt:**
```
org.elasticsearch.client.WarningFailureException: method [PUT], host [http://127.0.0.1:34431], URI [/my-index/_doc/1?pipeline=my_pipeline], status line [HTTP/1.1 201 Created]
Warnings: [index name [.monitoring-es-7-2022.03.10] starts with a dot '.', in the next major version, index names starting with a dot are reserved for hidden indices and system indices]
{"_index":"my-index","_id":"1","_version":1,"result":"created","_shards":{"total":2,"successful":1,"failed":0},"_seq_no":0,"_primary_term":1}

  at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:342)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:312)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:287)
  at org.elasticsearch.test.enrich.CommonEnrichRestTestCase.setupGenericLifecycleTest(CommonEnrichRestTestCase.java:122)
  at org.elasticsearch.test.enrich.CommonEnrichRestTestCase.testBasicFlowDate(CommonEnrichRestTestCase.java:148)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by joegallo**
Closes #84885

So the test failure from https://github.com/elastic/elasticsearch/issues/84885 is really interesting. What happens is that one test randomly fails due to the stacktrace below (which IIUC is an issue that @jbaiera is looking into in the general case), but then that kicks off a comedy of errors where everything fails:

```
org.elasticsearch.xpack.enrich.EnrichSecurityIT > testBasicFlowDate FAILED
    org.elasticsearch.client.WarningFailureException: method [PUT], host [http://127.0.0.1:34431/], URI [/my-index/_doc/1?pipeline=my_pipeline], status line [HTTP/1.1 201 Created]
    Warnings: [index name [.monitoring-es-7-2022.03.10] starts with a dot '.', in the next major version, index names starting with a dot are reserved for hidden indices and system indices]
    {"_index":"my-index","_id":"1","_version":1,"result":"created","_shards":{"total":2,"successful":1,"failed":0},"_seq_no":0,"_primary_term":1}
        at app//org.elasticsearch.client.RestClient.convertResponse(RestClient.java:342)
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:312)
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:287)
```

See, since that warning is attached to our request, that results in an exception being thrown, so the pipeline that that test just created is never deleted. Which means that the policy referencing that pipeline cannot be deleted and then subsequent tests can't put a new policy because there's already a non-deleted policy with the same name in place.

That is, once any test in this class fails (and so fails to clean up after itself) then all tests in that class from there on out will fail.

This PR fixes that.

### CODE REVIEW

File: x-pack/plugin/enrich/qa/common/src/main/java/org/elasticsearch/test/enrich/CommonEnrichRestTestCase.java [hunk 1]
```diff
@@ -34,22 +35,44 @@
...
-                try {
-                    client().performRequest(new Request("DELETE", "/" + sourceIndex));
-                } catch (ResponseException e) {
-                    // and that is ok
+    public void deletePipelinesAndPolicies() throws Exception {
+        {
+            // delete all pipelines
+            List<String> pipelines = new ArrayList<>();
+            try {
+                Map<String, Object> responseMap = toMap(adminClient().performRequest(new Request("GET", "/_ingest/pipeline")));
```
Comment (jbaiera) [inline x-pack/plugin/enrich/qa/common/src/main/java/org/elasticsearch/test/enrich/CommonEnrichRestTestCase.java:43]: We're not running this in parallel with any other tests that might be creating their own pipelines are we? Is that even something to worry about here?

**Review by jbaiera — APPROVED** [review]
LGTM pending green CI and tiny question

File: x-pack/plugin/enrich/qa/common/src/main/java/org/elasticsearch/test/enrich/CommonEnrichRestTestCase.java (hunk 1 above)
Comment (joegallo) [reply x-pack/plugin/enrich/qa/common/src/main/java/org/elasticsearch/test/enrich/CommonEnrichRestTestCase.java:43]: That's a fair point!

My thinking here was that we're already unconditionally deleting all the enrich policies in this cleanup, so we've already got a thread-safety resource-bashing issue. And while that's true, there's probably more tests out there in the world that use ingest pipelines than tests that use enrich policies, so this makes the potential damage bigger than it was before.

I think I can limit this damage pretty well, commit coming.

Comment (joegallo) [reply x-pack/plugin/enrich/qa/common/src/main/java/org/elasticsearch/test/enrich/CommonEnrichRestTestCase.java]: How about 9a67267f414f6023daa4f92ba4c07ce7604ecf2d, @jbaiera?

**Comment by iverase** [comment]
This test failed in 8.2 with the same error, are you planing in backporting it? 
https://gradle-enterprise.elastic.co/s/iqehqbkutx3wo/

**Comment by jbaiera** [comment]
@iverase This PR doesn't fix that error - it just makes it so that the entire suite doesn't fail. The root issue for that test failure is at https://github.com/elastic/elasticsearch/issues/85506

**Pull request merged**